__tests__/warehouse.test.js

```diff
@@ -0,0 +1,196 @@
+        );
+      }
+    });
+  });
+});
+
+describe("conflict between rudder set props and user set props", () => {
+  it("should override user set props with rudder prop", () => {
+    eventTypes.forEach(evType => {
+      if (evType === "alias" || evType === "group") return;
+
+      let i = input(evType);
+
+      let propsKey = "properties";
+      if (evType === "identify") {
+        propsKey = "traits";
+      }
+
+      rudderProps = _.compact(
+        rudderProperties.default.concat(rudderProperties[evType])
+      );
+
+      rudderProps.forEach(prop => {
+        i.message[propsKey][prop] = "test prop";
+      });
+
+      transformers.forEach(transformer => {
+        const received = transformer.process(i);
+        rudderProps.forEach(prop => {
+          expect(received[0].data[prop]).not.toBe("test prop");
+        });
+      });
+    });
+  });
+
+  it("should set data type of rudder prop and not user set prop", () => {
+    eventTypes.forEach(evType => {
+      if (evType === "alias" || evType === "group") return;
+
+      let i = input(evType);
+
+      let propsKey = "properties";
+      if (evType === "identify") {
+        propsKey = "traits";
+      }
+
+      transformers.forEach((transformer, index) => {
+        let sampleRudderPropKey = "id";
+        if (integrations[index] === "snowflake") {
+          sampleRudderPropKey = "ID";
+        }
+
+        i.message[propsKey][sampleRudderPropKey] = true;
+        const received = transformer.process(i);
+        // TODO: snowflake specific
+        expect(received[0].metadata.columns[sampleRudderPropKey]).not.toBe(
+          "bool"
+        );
+        expect(received[0].metadata.columns[sampleRudderPropKey]).toBe(
+          "string"
+        );
+      });
+    });
+  });
+
+  describe("handle reserved words", () => {
+    it("prepend underscore", () => {
+      eventTypes.forEach(evType => {
+        if (evType === "alias" || evType === "group") return;
+        let i = input(evType);
+
+        let propsKey = "properties";
+        if (evType === "identify") {
+          propsKey = "traits";
+        }
+
+        rudderProps = _.compact(
+          rudderProperties.default.concat(rudderProperties[evType])
+        );
+
+        rudderProps.forEach(prop => {
+          i.message[propsKey][prop] = "test prop";
+        });
+
+        transformers.forEach((transformer, index) => {
+          const reserverdKeywordsMap =
+            reservedANSIKeywordsMap[integrations[index].toUpperCase()];
+
+          i.message[propsKey] = Object.assign(
+            i.message[propsKey],
+            reserverdKeywordsMap
+          );
+
+          const received = transformer.process(i);
+
+          const out =
+            evType === "track" || evType === "identify"
+              ? received[1]
+              : received[0];
+
+          Object.keys(reserverdKeywordsMap).forEach(k => {
+            expect(out.metadata.columns).not.toHaveProperty(k.toLowerCase());
```

Comment:
Should we also check for `k` becoming `_${k}`

Reply:
added